src/pp_ast.ml

```diff
@@ -205,6 +216,23 @@ class lift_simple_val =
         ~lift_record:super#module_expr ~desc:mod_.pmod_desc
         ~attrs:mod_.pmod_attributes mod_
 
+    method! constant_desc (c : constant_desc) =
+      if config.full_consts then super#constant_desc c
+      else
+        match c with
+        | Pconst_integer (i, None) -> self#int (int_of_string i)
```

Comment:
I had something simpler in mind, i.e. still print the `_desc` part as we used but only skip the wrapping record.

I think printing the actual constant `_desc` is useful for users that need to discover how they are represented. What do you think?

Reply:
Yep, I've just used the existing logic for this